lib/mongoid/association/nested/many.rb

```diff
@@ -32,8 +32,10 @@ def build(parent, options = {})
           attributes.each do |attrs|
             if attrs.is_a?(::Hash)
               process_attributes(parent, attrs.with_indifferent_access)
-            else
+            elsif attrs[1].respond_to?(:with_indifferent_access)
               process_attributes(parent, attrs[1].with_indifferent_access)
+            else
+              process_attributes(parent, Hash[*attrs].with_indifferent_access)
```

Comment:
do we need more guards here in case it can't be cast to a hash?

Reply:
I think it's probably smart to check that array has a non-zero and even number of elements, before creating a hash from it.

We should probably also raise an exception if that's not the case, so that we never have data being silently ignored. A simple `ArgumentError` would probably suffice, indicating that the arguments aren't supported or something.